test/jdk/javax/sound/midi/Synthesizer/Receiver/bug6186488.java

```diff
@@ -21,52 +21,181 @@
 import javax.sound.midi.MidiMessage;
 import javax.sound.midi.MidiSystem;
+import javax.sound.midi.MidiUnavailableException;
+import javax.swing.JDialog;
+import javax.swing.SwingUtilities;
+import javax.swing.JTextArea;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.Timer;
+import javax.swing.JPanel;
+import javax.swing.WindowConstants;
 
-/**
+/*
  * @test
  * @bug 6186488
  * @summary Tests that software Java Syntesizer processed
  *          non-ShortMessage-derived messages
- * @run main/manual=yesno bug6186488
+ * @run main/manual bug6186488
  */
 public class bug6186488 {
-    public static void main(String[] args) throws Exception {
-        MidiDevice/*Synthesizer*/ synth = null;
+    private static final CountDownLatch countDownLatch = new CountDownLatch(1);
+    private static final int testTimeout = 300000;
+    private static volatile String testFailureMsg;
+    private static volatile boolean testPassed;
+    private static volatile boolean testFinished;
 
+    public static void main(String[] args) throws InterruptedException, InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> createAndShowTestDialog());
         try {
-            synth = MidiSystem.getSynthesizer();
-            //synth = MidiSystem.getMidiDevice(infos[0]);
+            if (!countDownLatch.await(testTimeout, TimeUnit.MILLISECONDS)) {
+                throw new RuntimeException(String.format("Test timeout '%d ms' elapsed.", testTimeout));
+            }
+            if (!testPassed) {
+                String failureMsg = testFailureMsg;
+                if ((failureMsg != null) && (!failureMsg.trim().isEmpty())) {
+                    throw new RuntimeException(failureMsg);
+                } else {
+                    throw new RuntimeException("Test failed.");
+                }
+            }
+        } catch (InterruptedException ie) {
+            throw new RuntimeException(ie);
+        } finally {
+            testFinished = true;
+        }
+    }
 
+    private static void pass() {
+        testPassed = true;
+        countDownLatch.countDown();
+    }
+
+    private static void fail(String failureMsg) {
+        testFailureMsg = failureMsg;
+        testPassed = false;
+        countDownLatch.countDown();
+    }
+
+    private static String convertMillisToTimeStr(int millis) {
+        if (millis < 0) {
+            return "00:00:00";
+        }
+        int hours = millis / 3600000;
+        int minutes = (millis - hours * 3600000) / 60000;
+        int seconds = (millis - hours * 3600000 - minutes * 60000) / 1000;
+        return String.format("%02d:%02d:%02d", hours, minutes, seconds);
+    }
+
+    private static void createAndShowTestDialog() {
+        String testInstruction = "This test verify that software Java Syntesizer processed non-ShortMessage-derived messages. \n" +
+                "Click on 'Start Test' button. If you listen a sound then test pass else test fail.";
```

Comment:
Probably it will be useful to ask the user to "Make sure that the speakers are connected and the volume is up."? Some other tests use this suggestion.

Reply:
Updated instructions to close any sound program and speaker to be connected and increase the volume up.